tests/dht/test_providers.nim

```diff
@@ -35,34 +39,31 @@ proc initProvidersNode(
   let d = initDiscoveryNode(rng, privKey, address, bootstrapRecords)
   newProvidersProtocol(d)
 
-proc toPeerRecord(p: ProvidersProtocol) : PeerRecord =
-  ## hadle conversion between the two worlds
-
-  #NodeId is a keccak-256 hash created by keccak256.digest and stored as UInt256
-  let discNodeId = p.discovery.localNode.id
-  ## get it back to MDigest form
-  var digest: MDigest[256]
-  digest.data = discNodeId.toBytesBE
-  ## get into a MultiHash
-  var mh = MultiHash.init(multiCodec("keccak-256"), digest).orError(HashError).get()
-  result = PeerRecord.init(
-    peerId = PeerId.init(mh.data.buffer).get,
+proc toSignedPeerRecord(privKey: crypto.PrivateKey) : SignedPeerRecord =
+  ## handle conversion between the two worlds
+
+  let pr = PeerRecord.init(
+    peerId = PeerId.init(privKey.getPublicKey.get).get,
     seqNo = 0,
     addresses = @[])
```

Comment:
This works for our test, but later we better initialize this with the actual address.

Reply:
Have added some example addresses for now.